File: backend/src/main/java/com/example/backend/service/CommentService.java
package com.example.backend.service;

import com.example.backend.model.request.CommentRequest;
import com.example.backend.model.respon.CommentResponse;

public interface CommentService {
    // thêm bình luận
    CommentResponse createComment(Long userId, CommentRequest request);

    // chỉnh sửa bình luận
    CommentResponse updateComment(Long userId, Long commentId, String content);

    // Xóa mềm bình luận (cho phép chủ bình luận hoặc chủ bài viết xóa)
    Boolean deleteComment(Long userId, Long commentId);

    // Lấy danh sách bình luận gốc của bài viết (có phân trang)
    org.springframework.data.domain.Page<CommentResponse> getCommentsByPostId(Long postId, int page, int size);

    // Lấy danh sách bình luận trả lời (reply) của một bình luận cha
    org.springframework.data.domain.Page<CommentResponse> getRepliesByCommentId(Long commentId, int page, int size);
}

File: frontend/src/service/impl/CommentService.ts
import { http } from "../http/http";
import type { CommentRequest } from "../../utils/RequestType";
import type { ApiDataResponse, Page, CommentResponse } from "../../utils/ResponType";

export const createComment = (currentUserId: number, request: CommentRequest) => {
    return http.post<ApiDataResponse<CommentResponse>>(`/comments?currentUserId=${currentUserId}`, request);
};

export const updateComment = (currentUserId: number, commentId: number, content: string) => {
    return http.put<ApiDataResponse<CommentResponse>>(`/comments/${commentId}?currentUserId=${currentUserId}`, content, {
        headers: {
            "Content-Type": "text/plain",
        },
    });
};

export const deleteComment = (currentUserId: number, commentId: number) => {
    return http.delete<ApiDataResponse<boolean>>(`/comments/${commentId}?currentUserId=${currentUserId}`);
};

export const getCommentsByPostId = (postId: number, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<CommentResponse>>>(`/comments/posts/${postId}?page=${page}&size=${size}`);
};

export const getRepliesByCommentId = (commentId: number, page: number = 0, size: number = 20) => {
    return http.get<ApiDataResponse<Page<CommentResponse>>>(`/comments/${commentId}/replies?page=${page}&size=${size}`);
};
